tests/testthat/test-forecasters-data.R

```diff
@@ -7,79 +7,105 @@ forecasters <- tibble::tribble(
   scaled_pop, list(1, TRUE), list("ahead", "pop_scaling"), "scaled_pop",
   scaled_pop, list(1, FALSE), list("ahead", "pop_scaling"), "scaled_pop",
   flatline_fc, list(1), list("ahead"), "flatline_fc",
-  smoothed_scaled, list(1, list(c(0,7,14), c(0))), list("ahead", "lags"), "smoothed_scaled"
+  smoothed_scaled, list(1, list(c(0, 7, 14), c(0))), list("ahead", "lags"), "smoothed_scaled"
 )
 expects_nonequal <- c("scaled_pop", "smoothed_scaled")
-synth_mean <- 25
-synth_sd <- 2
-tiny_sd <- 1.0e-5
-simple_dates <- seq(as.Date("2012-01-01"), by = "day", length.out = 40)
-# technically white noise, but with a variance that is miniscule
-constant <- epiprocess::as_epi_archive(tibble(
-  geo_value = "al",
-  time_value = simple_dates,
-  version = simple_dates,
-  a = synth_mean
-))
-
 # wrap a call that is made quite frequently
 # n_training_pad is set to avoid warnings from the trainer
-get_pred <- function(dataset,
-                     ii, outcome = "a", extra_sources = "") {
-  res <- slide_forecaster(
+default_slide_forecaster <- function(dataset,
```

Comment:
Can you bundle all the comments about this function into a single cohesive docstring above it? Like "this function returns a commonly used slide forecaster and provides options to silence common annoying warnings from linreg". I personally prefer to read a single paragraph explaining a thing instead of piecing it together from non-linear reading.